---
title: "class15DifferentialExpressionAnalysis"
author: "Kiana Miyamoto"
date: "11/15/2019"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##About our Input Data
The data for for hands-on session comes from GEO entry: GSE37704, which is associated with the following publication:

Trapnell C, Hendrickson DG, Sauvageau M, Goff L et al. "Differential analysis of gene regulation at transcript resolution with RNA-seq". Nat Biotechnol 2013 Jan;31(1):46-53. PMID: 23222703
The authors report on differential analysis of lung fibroblasts in response to loss of the developmental transcription factor HOXA1. 

```{r}
##Importing the data
metaFile<-"GSE37704_metadata.csv"
countFile<-"GSE37704_featurecounts.csv"
#Import metadata and take a peek
colData = read.csv(metaFile,row.names=1)
head(colData)
#Import countdata
countData = read.csv(countFile,row.names=1)
head(countData)
```

Looks like we nee dto remove the first column in countData, the "length" col
```{r}
countData<-countData[,-1]
```

Remove genes with zero counts across all experiments
```{r}
countData<-countData[rowSums(countData)> 0,]
countData
```

How many genes are we left with?
```{r}
nrow(countData)
```
PCA
The first analysis step is usually always to plot the data but here we have 15K genes. How do we plot this - PCA to the rescue!
```{r}
PC<-prcomp(t(countData))
plot(PC)
```

```{r}
summary(PC)
```

```{r}
attributes(PC)
mycols<- c(rep("blue",3), rep("red", 3))
plot(PC$x[,1:2], col=mycols)
```

```{r}
library(DESeq2)
```

```{r}
dds = DESeqDataSetFromMatrix(countData=countData,
                             colData=colData,
                             design=~condition)
dds = DESeq(dds)
```

```{r}
dds
```

```{r}
res = results(dds)
summary(res)
```
##Volcano Plot Summary
```{r}
plot(res$log2FoldChange, -log(res$padj))
```
```{r}
# Make a color vector for all genes
mycols <- rep("gray", nrow(res) )

# Color red the genes with absolute fold change above 2
mycols[ abs(res$log2FoldChange) > 2 ] <- "red"

# Color blue those with adjusted p-value less than 0.01
#  and absolute fold change more than 2
inds <- res$padj < 0.01 & (abs(res$log2FoldChange) > 2 )
mycols[ inds ] <- "blue"
plot( res$log2FoldChange, -log(res$padj), col=mycols, xlab="Log2(FoldChange)", ylab="-Log(P-value)" )
```

```{r}
library("AnnotationDbi")
library("org.Hs.eg.db")
```

```{r}
columns(org.Hs.eg.db)

res$symbol = mapIds(org.Hs.eg.db,
                    keys=row.names(res), # where are your gene ids?
                    keytype="ENSEMBL", # what format are your ids?
                    column="SYMBOL", # what NEW ids format do you want?
                    multiVals="first")

res$entrez = mapIds(org.Hs.eg.db,
                    keys=row.names(res),
                    keytype="ENSEMBL",
                    column="ENTREZID",
                    multiVals="first")

res$name =   mapIds(org.Hs.eg.db,
                    keys=row.names(res),
                    keytype="ENSEMBL",
                    column="GENENAME",
                    multiVals="first")

head(res, 10)
```

```{r}
##BiocManager::install(c("pathview", "gage", "gageData"))
```

```{r}
library(pathview)
```

```{r}
library(gage)
library(gageData)
```

```{r}
data(kegg.sets.hs)
data(sigmet.idx.hs)
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]
head(kegg.sets.hs,3)
```
The main gage() function requires a named vector of fold changes, where the names of the values are the Entrez gene IDs.

Note that we used the mapIDs() function above to obtain Entrez gene IDs (stored in res$entrez) and we have the fold change results from DESeq2 analysis (stored in res$log2FoldChange).
```{r}
foldchanges= res$log2FoldChange
names(foldchanges) = res$entrez
head(foldchanges)
```
Now, letâ€™s run the gage pathway analysis.
```{r}
keggres = gage(foldchanges, gsets=kegg.sets.hs)
attributes(keggres)
```
Now lets look at the object returned from gage().

```{r}
head(keggres$less)
```
Now, let's try out the pathview() function from the pathview package to make a pathway plot with our RNA-Seq expression results shown in color.
To begin with lets manually supply a pathway.id (namely the first part of the "hsa04110 Cell cycle") that we could see from the print out above.
```{r}
pathview(gene.data=foldchanges, pathway.id="hsa04110")
```

```{r}
# A different PDF based output of the same data
pathview(gene.data=foldchanges, pathway.id="hsa04110", kegg.native=FALSE)
```

![My first pathway](./hsa04110.pathview.png)

##Gene Ontology (GO)
```{r}
data(go.sets.hs)
data(go.subs.hs)

# Focus on Biological Process subset of GO
gobpsets = go.sets.hs[go.subs.hs$BP]

gobpres = gage(foldchanges, gsets=gobpsets, same.dir=TRUE)

lapply(gobpres, head)
```

